apps/emqx_bridge/src/emqx_bridge_api.erl

```diff
@@ -541,7 +541,7 @@ schema("/bridges_probe") ->
     case emqx_dashboard_swagger:filter_check_request_and_translate_body(Request, RequestMeta) of
         {ok, #{body := #{<<"type">> := ConnType} = Params}} ->
             Params1 = maybe_deobfuscate_bridge_probe(Params),
-            case emqx_bridge_resource:create_dry_run(ConnType, maps:remove(<<"type">>, Params1)) of
+            try emqx_bridge_resource:create_dry_run(ConnType, maps:remove(<<"type">>, Params1)) of
                 ok ->
                     ?NO_CONTENT;
                 {error, #{kind := validation_error} = Reason} ->
```

Comment:
how does the code end up here before this PR ? I cannot seem to find a type check called anywhere.

Reply:
I assume it was this code inside `create_dry_run`:

```erlang
            try
                ParseConf = parse_confs(bin(Type), TmpName, ConfNew),
                emqx_resource:create_dry_run_local(bridge_to_resource_type(Type), ParseConf)
            catch
                %% validation errors
                throw:Reason ->
                    {error, Reason}
```

Reply:
But indeed there seems to be no type checking there either.  :thinking: